File: Java/Średnia arytmetyczna.java
import java.util.Arrays;
import java.util.Scanner;

public class Main 
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);
        int n = input.nextInt();
        int[] numbers = new int[n];
        for (int i = 0; i < n; i++) 
            numbers[i] = input.nextInt();
        Arrays.sort(numbers);
        int result = (n % 2 != 0) ? numbers[n / 2] : numbers[(n + 1) / 2];
        System.out.println(result);
        input.close();
    }
}

File: C++/Średnia arytmetyczna.cpp
#include <vector>
#include <iostream>
#include <algorithm>

int main() 
{
    int n;
    std::cin >> n;
    std::vector<int> numbers(n);
    for (int i = 0; i < n; ++i)
        std::cin >> numbers[i];
    std::sort(numbers.begin(), numbers.end());
    int result = (n % 2 != 0) ? numbers[n / 2] : numbers[(n + 1) / 2];
    std::cout << result;
    return 0;
}